auth/src/main/res/layout/fui_confirmation_code_layout.xml

```diff
@@ -6,100 +6,106 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent">
 
-    <RelativeLayout
+    <android.support.constraint.ConstraintLayout
         style="@style/FirebaseUI.WrapperStyle"
-        android:layout_width="match_parent"
         android:layout_height="wrap_content">
 
         <TextView
-            android:id="@+id/dgts_titleText"
+            android:id="@+id/header"
             style="@style/FirebaseUI.Text.T08"
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
-            android:text="@string/fui_enter_confirmation_code" />
+            android:text="@string/fui_enter_confirmation_code"
+            app:layout_constraintTop_toTopOf="parent" />
 
         <TextView
             android:id="@+id/edit_phone_number"
             style="@style/FirebaseUI.Text.T08"
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
-            android:layout_below="@id/dgts_titleText"
             android:textAppearance="?android:attr/textAppearanceMedium"
             android:textColor="@color/fui_linkColor"
+            app:layout_constraintTop_toBottomOf="@id/header"
             tools:ignore="SelectableText" />
 
-        <com.firebase.ui.auth.ui.phone.SpacedEditText
-            android:id="@+id/confirmation_code"
+        <android.support.design.widget.TextInputLayout
```

Comment:
@SUPERCILEX did you tested on any device with API lower than Lollipop? `TexInputLayout` is pretty UI improvement, bu it may looks weird on older devices :-)